Entity/ProductInterface.php

```diff
@@ -411,11 +412,37 @@ public function removeProductAttribute(ProductAttribute $productAttributes);
     /**
      * Get attributes.
      *
-     * @return \Doctrine\Common\Collections\Collection
+     * @return ArrayCollection
      */
     public function getProductAttributes();
 
     /**
+     * @return ArrayCollection
+     */
+    public function getVariantAttributes();
+
+    /**
+     * @param Attribute $attribute
+     *
+     * @return self
+     */
+    public function addVariantAttribute(Attribute $attribute);
+
+    /**
+     * @param Attribute[] $attributes
+     *
+     * @return self
```

Comment:
i am a bit fan of `@return $this` which implicit makes it clear which instance is returned. `@return self` declares that you will return the same type not the same instance.